### PR TITLE
Adding a ping on others servers in backoffice

### PR DESCRIPTION
**Pull request by olivierteboul**
This PR adds a systematic ping to the www and message servers from the backoffice and always gives the status of those two in the nav bar (for admins only)

### CODE REVIEW

**Comment by olivierteboul** [comment]
image

**Comment by rth** [comment]
Nice! Just to confirm: so each of those handlers now need to be async, otherwise for each request we will be waiting for the pinged server to respond?

**Comment by olivierteboul** [comment]
i don t think it would hang, but i am not sure what would happen. I can give it a try to double check. But having async handlers is not a big deal. on the contrary you get more than you loose

File: icubam/backoffice/handlers/base.py [hunk 1]
```diff
@@ -13,9 +14,25 @@ def initialize(self, config, db):
...
+        result[key] = {'status': resp.code == 200, 'since': resp.body}
+      except:
+        result[key] = {'status': False, 'since': '-'}
+        continue
+    return result
 
-  def render(self, path, **kwargs):
-    super().render(path, this_user=self.user, **kwargs)
+  async def render(self, path, **kwargs):
+    pings = await self.ping() if self.user and self.user.is_admin else {}
```
Comment (sgirgin) [inline icubam/backoffice/handlers/base.py:34]: Should this be handled by a periodic callback instead? It is easy to define one in tornado it seems and will be better than pinging at each request.

Comment (olivierteboul) [reply icubam/backoffice/handlers/base.py:34]: sure. let me give it a try

Comment (sgirgin) [reply icubam/backoffice/handlers/base.py:34]: Tentatively approving as discussed. You can change it later.

Comment (olivierteboul) [reply icubam/backoffice/handlers/base.py]: done !

**Pull request merged**